### PR TITLE
349 plotting does not support custom dimords

### PR DESCRIPTION
**Pull request by tensionhead**
This PR actually addreses a bunch of minor bugs, that appeared during manual testing:

- `spectralDTypes` in `shared.const_def` now contain all supported outputs
- multi-taper parameters now are aware of the maximal smoothing bw possible (edge case)
- fixed dimord for AnalogData, for other datatypes it is most likely not needed, as the frontend results always have the default dimord (how would you get a `SpectralData` with non-default dimord? ..we don't support reading FT spectral data atm anyways)
- now for `output != 'pow'` there is no general log10, so imaginary/real parts of spectra/coherence can also be plotted nicely (they of course can have negative values)
- cut back on the test helper selection generation, one random combination per toi/toilim setting is enough 
- also addressed #357, I hope there are no undesired side effects.. tests are not complaining though

Author Guidelines
-----------------
- [x] Is the change set **< 600 lines**?
- [x] Was the code checked for memory leaks/performance bottlenecks?
- [x] Is the code running locally **and** on the ESI cluster?
- [x] Is the code running on all supported platforms?

Reviewer Checklist
------------------
- [x] Are testing routines present?
- [x] Do **parallel** loops have a set length and correct termination conditions?
- [x] Do objects in the global package namespace perform proper parsing of their input? 
- [x] Do code-blocks provide novel functionality, i.e., no re-factoring using builtin/external packages possible?
- [x] Code layout
  - [x] Is the code PEP8 compliant?
  - [x] Does the code adhere to agreed-upon naming conventions?
  - [x] Are keywords clearly named and easy to understand?
  - [x] No commented-out code?
- [x] Are all docstrings complete and accurate?
- [x] Is the CHANGELOG.md up to date?

### CODE REVIEW

File: syncopy/plotting/mp_plotting.py [hunk 1]
```diff
@@ -63,7 +66,7 @@ def plot_AnalogData(data, shifted=True, **show_kwargs):
         return
 
     elif nAx > pltConfig['mMaxAxes']:
-        SPYWarning("Please select max. {pltConfig['mMaxAxes']} channels for a multipanelplot!")
+        SPYWarning(f"Please select max. {pltConfig['mMaxAxes']} channels for a multipanelplot!")
```
Comment (dfsp-spirit) [inline syncopy/plotting/mp_plotting.py:69]: oh yes, forgetting that `f` is a common bug.

File: syncopy/tests/test_plotting.py [hunk 2]
```diff
@@ -23,13 +23,15 @@ class TestAnalogPlotting():
     nSamples = 300
     adata = synth_data.AR2_network(nTrials=nTrials,
                                    AdjMat=np.zeros(nChannels),
-                                   nSamples=nSamples)
+                                   nSamples=nSamples,
+                                   seed=helpers.test_seed)
```
Comment (dfsp-spirit) [inline syncopy/tests/test_plotting.py:27]: Nice, it's a good idea to have one seed in a common place.

**Comment by dfsp-spirit** [comment]
Thanks for this great contribution!

I have one general question that we should discuss soon maybe, that came up during the review, but is not specific to this PR: when do we give warnings, and when errors? E.g., the plotting functions just print something (technically not even a warning) without plotting anything in case of invalid input args.

Throwing an error would also make it easier to test for the exceptions, and to increase coverage.

**Comment by tensionhead** [comment]
> Thanks for this great contribution!
> 
> I have one general question that we should discuss soon maybe, that came up during the review, but is not specific to this PR: when do we give warnings, and when errors? E.g., the plotting functions just print something (technically not even a warning) without plotting anything in case of invalid input args.
> 
> Throwing an error would make it easier to test for the exceptions, and to increase coverage. It may have other downsides, though.

Yes absolutely, this discussion is much needed! For the current state of the code base, I did not want to 'kill' an analysis script by raising an exception just because some plotting did not go down as expected..

**Pull request merged**